hazelcast/proxy/replicated_map.py

```diff
@@ -78,21 +78,33 @@ def add_entry_listener(
             A registration id which is used as a key to remove the listener.
         """
         if key and predicate:
```

Comment:
The `key` comparisons in this part don't look correct to me. What if `key == 0`?